server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -163,8 +165,9 @@ public void beforeRefresh() throws IOException {}
      */
     @Override
     public void afterRefresh(boolean didRefresh) {
-
-        if (didRefresh || remoteDirectory.getSegmentsUploadedToRemoteStore().isEmpty()) {
+        if ((remoteSegmentMetadata != null && remoteSegmentMetadata.getPrimaryTerm() != indexShard.getOperationPrimaryTerm())
```

Comment:
What does this change do?

Reply:
* We need to call refresh after shard gets started because replicationTracker.primaryMode() is not true till the node gets in started state, if we close index and open.
* we need to add condition for different primary Term for scenarios when we are trying to restore snapshot in the same index because in that case, even though we are going to afterRefresh method, we are still not going to syncSegments method because neither didRefresh is true nor segmentsInRemoteStore are empty.

Reply:
Once `remoteSegmentMetadata.getPrimaryTerm() != indexShard.getOperationPrimaryTerm())` is true, won't it always be true? In what scenario, it become false?